linode/manager
# M3-5452: Add support for Autoscaling Node Pools

## Description
Adds support for Autoscaling a Node Pool on a Kubernetes Cluster's summary page

## How to test
Go to `kubernetes/clusters/<cluser_id>/summary`, ensure the `autoscaler` feature flag is on

- You should see an `Autoscale Pool` button before the Resize Pool button
    - If the Autoscaler is on, you should see min/max values next to the autoscale pool button
    - If the Autoscaler is off, you should not see min/max values next to the autoscale pool button
- Clicking on Autoscale Pool should display a dialog
- The Save Changes button should be disabled until there has been a change in the form and there are no errors
- If Autoscaler is off, the Min/Max input fields should be disabled
- If the Minimum value is not between 1-99 or greater than the Maximum value, there should be an error displayed under the input field
- If the Maximum value is not between 1-100, there should be an error displayed under the input field
- If the Maximum value is less than the current Max, then a warning notice should display
    - Clicking on `resize` in the warning should close the modal and open the resize drawer for that pool
- Clicking Save Changes should make a request to `lke/clusters/<cluster_id>/pools/<pool_id>`
    - Any API errors should be displayed as a Notice error in the dialog
    - A successful request should close the dialog, pop a success toast and update the Min/Max values next to the Autoscale Pool button (might take a few seconds to update)
